docs/docs/03-light-clients/04-wasm/09-migrations.md

```diff
@@ -9,6 +9,94 @@ slug: /ibc/light-clients/wasm/migrations
 
 This guide provides instructions for migrating 08-wasm versions.
 
+## From v0.2.0+ibc-go-v8.3-wasmvm-v2.0 to v0.3.0-ibc-go-v8.3-wasmvm-v2.0
+
+### Contract developers
+
+The `v0.3.0` release of 08-wasm for ibc-go `v8.3.x` and above introduces a breaking change for client contract developers.
+
+The contract API `SudoMsg` payloads `VerifyMembershipMsg` and `VerifyNonMembershipMsg` have been modified. 
+The encoding of the `Path` field of both structs has been updated from `v1.MerklePath` to `v2.MerklePath` to support proving values stored under keys which contain non-utf8 encoded symbols. 
+
+As a result, the `Path` field now contains a `MerklePath` composed of `key_path` of `[][]byte` as opposed to `[]string`. The JSON field `path` containing `key_path` of both `VerifyMembershipMsg` and `VerifyNonMembershipMsg` structs will now marshal elements as base64 encoded bytestrings. See below for example JSON diff.
+
+```diff
+{
+  "verify_membership": {
+    "height": {
+      "revision_height": 1
+    },
+    "delay_time_period": 0,
+    "delay_block_period": 0,
+    "proof":"dmFsaWQgcHJvb2Y=",
+    "path": {
++      "key_path":["L2liYw==","L2tleS9wYXRo"]
+-      "key_path":["/ibc","/key/path"]
+    },
+    "value":"dmFsdWU="
+  }
+}
+```
+
+A migration is required for existing 08-wasm client contracts in order to correctly handle the deserialisation of `key_path` from `[]string` to `[][]byte`.
+Contract developers should familiarise themselves with the migration path offered by 08-wasm [here](./05-governance.md#migrating-an-existing-wasm-light-client-contract).
+
+An example of the required changes in a client contract may look like:
+
+```diff
+#[cw_serde]
+pub struct MerklePath {
++   #[schemars(with = "String")]
++   #[serde(with = "Base64", default)]
++   pub key_path: Vec<Bytes>,
+-   pub key_path: Vec<String>,
+}
```

Comment:
@srdtrk, could you confirm if this is correct unmarshalling syntax for `[]string` -> `[][]byte`? I can also modify the composable-ibc grandpa-cw contract and try to test this

Reply:
This wouldn't work

Reply:
What would be the right way to do it, @srdtrk?

Reply:
We've decided to replace the usage of Bytes with `cosmwasm_std::Binary`

Reply:
https://docs.rs/cosmwasm-std/2.0.4/cosmwasm_std/struct.Binary.html

> Binary is a wrapper around Vec to add base64 de/serialization with serde. It also adds some helper methods to help encode inline.

> This is only needed as serde-json-{core,wasm} has a horrible encoding for Vec. See also https://github.com/CosmWasm/cosmwasm/blob/main/docs/MESSAGE_TYPES.md.